### PR TITLE
Allow "disable mouse button" setting to apply to touch as well

### LINKED ISSUE #12019: No iPad Magic Keyboard input when using touch on iPad Pro
Keypresses are only detected when using the integrated trackpad or external mouse. When playing with touch or Apple Pencil, keypresses are ignored from the iPad Magic Keyboard. This was previously working many months ago, but has not started working again as of 2021.312.0. Device is iPad Pro 12.9 2018.

### PR DESCRIPTION
**Pull request by peppy**
PR for proposal. Not sure if these options should be joined together like this (but at the same time, I'm not sure the original option should even exist, rather than having users unbind their mouse buttons).

If this is to stick, we may want to update the localisation text on the option to also mention touch (and tablet?).

- [x] Depends on ppy/osu-framework#4983
- Closes https://github.com/ppy/osu/issues/12019

### CODE REVIEW

**Comment by bdach** [comment]
> but at the same time, I'm not sure the original option should even exist, rather than having users unbind their mouse buttons

In two minds about this. From a code perspective that definitely is a compelling solution but there are multiple bindings that would need to be unbound (left/right action for osu!, quick exit hotkey which is M4 on default on mice that have that, etc.) and another rather common use case is tablet users who may be pressing mouse buttons by pressing the stylus into the tablet, or other buttons placed on the stylus by accident during gameplay, but still desire "tap to click" and stylus buttons to work *outside* of gameplay.

So in summary I would weakly agree with this way forward but I could still be swayed the other way if there is a compelling argument for that (other than the usual "less settings" one, but I'm less inclined to agree with that given that it's already a setting now).

**Review by smoogipoo — APPROVED** [review]
